plugins/flytekit-kf-pytorch/flytekitplugins/kfpytorch/task.py

```diff
@@ -103,13 +106,18 @@ class PyTorch(object):
         worker: Configuration for the worker replica group.
         run_policy: Configuration for the run policy.
         num_workers: [DEPRECATED] This argument is deprecated. Use `worker.replicas` instead.
+        increase_shared_mem (bool): PyTorch uses shared memory to share data between processes. If torch multiprocessing is used
+            (e.g. for multithreaded data loaders) the default shared memory segment size that the container runs with might not be enough
+            and and one might have to increase the shared memory size. This option configures the task's pod template to mount
+            an `emptyDir` volume with medium `Memory` to to `/dev/shm`.
     """
 
     master: Master = field(default_factory=lambda: Master())
     worker: Worker = field(default_factory=lambda: Worker())
     run_policy: Optional[RunPolicy] = field(default_factory=lambda: None)
     # Support v0 config for backwards compatibility
     num_workers: Optional[int] = None
+    increase_shared_mem: bool = True
```

Comment:
While the tracking issue only discussed the elastic pytorch plugin, I'd make the case that the non-elastic task type benefits from increasing the shared memory by default as well.
As soon as you use data loaders with multiple worker processes, you very likely need this. In the past, I have added the shared memory volume to the flyte pod template configured in the flyte namespace which is applied to all task pods. This has never caused a problem.

Reply:
Can we have another parameter to configure the size limit of `V1EmptyDirVolumeSource`?

Reply:
@thomasjpfan have you been in a situation where this was required?
I've run hundreds, if not thousands, of these jobs and have never had a problem not setting the size limit. I'm not opposed to setting the size limit if it is required but if not, I'd prefer to avoid an unnecessary flag here.

Reply:
I always explicitly set it. I am okay with not having another flag if we document what the behavior is.

Reply:
https://github.com/flyteorg/flytekit/pull/2543/commits/927e9012af7aa9a2a9f6d878a59ed2ac708c4ad2